addons/survey/models/survey.py

```diff
@@ -0,0 +1,1159 @@
+            'type': 'ir.actions.act_url',
+            'name': "View Answers",
+            'target': 'self',
+            'url': '%s/%s' % (user_input['print_url'], user_input['token'])
+        }
+
+    @api.multi
+    def action_survey_results(self):
+        ''' Open the website page with the survey results '''
+        self.ensure_one()
+        return {
+            'type': 'ir.actions.act_url',
+            'name': "Survey Results",
+            'target': 'self',
+            'url': self.read(['result_url'])[0]['result_url']
+        }
+
+    @api.one
+    def copy_data(self, default=None):
+        raise UserError(_('You cannot duplicate this element!'))
+
+    @api.model
+    def do_clean_emptys(self, automatic=False):
+        ''' Remove empty user inputs that have been created manually
+            (used as a cronjob declared in data/survey_cron.xml) '''
+        empty_user_inputs = self.search([('input_type', '=', 'manually'),
+                                                     ('state', '=', 'new'),
+                                                     ('date_create', '<', fields.Datetime.to_string((datetime.datetime.now() - datetime.timedelta(hours=1))))])
+        if empty_user_inputs:
+            empty_user_inputs.unlink()
+
+class SurveyUserInputLine(models.Model):
+    _name = 'survey.user_input_line'
+    _description = 'Survey User Input Line'
+    _rec_name = 'date_create'
+
+    user_input_id = fields.Many2one('survey.user_input', string='User Input',
+                                     ondelete='cascade', required=True)
+    question_id = fields.Many2one('survey.question', string='Question',
+                                   ondelete='restrict', required=True)
+    page_id = fields.Many2one('survey.page', related='question_id.page_id',
+                              string="Page", readonly=True)
+    survey_id = fields.Many2one('survey.survey', related='user_input_id.survey_id',
+                                string='Survey', store=True, readonly=True)
+    date_create = fields.Datetime(string='Create Date', required=True, default=fields.Datetime.now)
+    skipped = fields.Boolean(default=False)
+    answer_type = fields.Selection([('text', 'Text'),
+                                     ('number', 'Number'),
+                                     ('date', 'Date'),
+                                     ('free_text', 'Free Text'),
+                                     ('suggestion', 'Suggestion')],
+                                    string='Answer Type')
+    value_text = fields.Char(string="Text answer")
+    value_number = fields.Float(string="Numerical answer")
+    value_date = fields.Datetime(string="Date answer")
+    value_free_text = fields.Text(string="Free Text answer")
+    value_suggested = fields.Many2one('survey.label', string="Suggested answer")
+    value_suggested_row = fields.Many2one('survey.label', "Row answer")
+    quizz_mark = fields.Float(string="Score given for this choice")
```

Comment:
actually we overrided create/write method to set quizz_mark value base on "value_suggested" field. so I propose to convert into compute field instead of override create/write method